Objects/typeobject.c

```diff
@@ -9155,6 +9161,10 @@ releasebuffer_call_python(PyObject *self, Py_buffer *buffer)
         PyObject_CallMethodNoArgs(mv, &_Py_ID(release));
     }
     Py_DECREF(mv);
+end:
+    assert(!PyErr_Occurred());
```

Comment:
This needs to checked, not asserted. `PyObject_CallMethodNoArgs(mv, &_Py_ID(release));` above can raise an exception.

Reply:
Good catch, I missed it, https://github.com/python/cpython/pull/104417 addresses it